WordPress/gutenberg
# Fix converting caption shortcode with link

## Description
Avoids a duplicate image when converting from a classic block.

Fixes #12310.

### Testing Instructions
- Create a classic post
- Insert an image.
- Add a caption, AND link to the media file
- Publish the classic post
- Edit the post in the block editor
- Convert the classic block to blocks
<img width="338" alt="screen shot 2018-12-09 at 11 49 27 am" src="https://user-images.githubusercontent.com/1270189/49700754-85a41280-fba8-11e8-83f8-f21fda37374e.png">
- In master we see:
<img width="787" alt="screen shot 2018-12-09 at 11 48 14 am" src="https://user-images.githubusercontent.com/1270189/49700743-5ee5dc00-fba8-11e8-8555-00e3e273ecf4.png">
In the branch we see:
<img width="552" alt="screen shot 2018-12-09 at 11 51 14 am" src="https://user-images.githubusercontent.com/1270189/49700776-c7cd5400-fba8-11e8-98f7-6cc3377a9220.png">

## How has this been tested?
See #12310. E2e tests have been added.

## Screenshots 

## Types of changes
Bug fix
## Checklist:
